### DISCUSSION TITLE
How to use migrations?

### DISCUSSION THREAD

**Discussion by josias-r** [prisma/prisma#19243] [Q&A] [answered]
I setup a brand new prisma project, but I'm failing to use migrations, only the initial migration worked, everything after that breaks :(

[detailed description](https://stackoverflow.com/questions/76243859/prisma-migrate-not-working-due-to-schema-not-empty-but-schema-is-empty-error-p)

**Comment by ludralph** [comment]
Hi @josias-r 👋 

One possible reason for this error is that you probably set up Prisma on an existing database. Take a look at this [answer](https://stackoverflow.com/a/74295122/1645620) on how to resolve it.

You can also choose to start with a fresh database and delete the migration folder then follow this [guide](https://www.prisma.io/docs/concepts/components/prisma-migrate/get-started) to learn how to use migrations.

**Reply by josias-r** [reply]
@ludralph That is the point, I did not, at least from my understanding what a "fresh" DB means. What I did:

1. Setup a fresh database - ✅
2. Setup a fresh prisma schema - ✅
3. Run the init migration - ✅
4. Make changes to the schema  - ✅
5. Generate a migration with the changes (still on the database where the first migration was applied to) ❌

This makes me feel like that just one initial migration already makes the DB "an existing DB". If I am not supposed to run migrations for an existing (prisma based since the start) DB, then what is the point of these migrations? 🤔

**Reply by ludralph** [reply]
Hi @josias-r, I have tried reproducing this issue but can't seem to do so. I can create multiple migrations so there must be something going on in your environment. Can you try resetting the database? This [guide](https://www.prisma.io/docs/concepts/components/prisma-migrate/migrate-development-production#reset-the-development-database) should help with it.

**Reply by josias-r** [reply]
@ludralph 
⚙️ `prisma migrate reset`
> ✔ Are you sure you want to reset your database? All data will be lost. … yes
> 
> Database reset successful

⚙️ `prisma migrate dev --name init`
> Applying migration `20230515134608_init`
> 
> The following migration(s) have been created and applied from new schema changes:
> 
> migrations/
>   └─ 20230515134608_init/
>     └─ migration.sql
> 
> Your database is now in sync with your schema.

⚙️ _update the prisma schema (adding a new optional String column to one of the entities)_

⚙️ `prisma migrate dev --name testmigrate`
> Error: P3005
> 
> The database schema is not empty. Read more about how to baseline an existing production database: https://pris.ly/d/migrate-baseline

I am working with a brand new postgres DB on vercel and I am using their recommended config:
![Screenshot 2023-05-15 at 15 52 28](https://github.com/prisma/prisma/assets/11424820/24579f40-f2a7-4e8f-bb81-2d5ed8bf36d8)

**Reply by ludralph** [reply, marked as answer]
Hi @josias-r, I'm able to reproduce the exact same error on vercel-postgres when running subsequent migrations. I also found an open [issue](https://github.com/vercel/storage/issues/114) on Vercel's GitHub page for this same problem.

**Reply by josias-r** [reply]
@ludralph Alright, thank you for the clarification. As someone who setup Prisma for the first time, I was really confused if I am misunderstanding how migrations are supposed to work. I will subscribe to that issue :D

**Reply by Radicitus** [reply]
Also having this issue, and it's a big one for sure. I understand Vercel Postgres is still in beta, but it's crazy migrations aren't working yet! Hoping my comment helps get this issue resolved faster.

**Reply by janpio** [reply]
Hey everyone, Vercel Postgres now supports Prisma Migrate better, so you can remove the `shadowDatabaseUrl` from your Prisma schema. Only `url` and `directUrl` is still needed, and Migrate should take care of the rest under the hood 🪄

Can you please let me know if this works for you? Then we can push to get the documentation updated accordingly. Thanks!

**Reply by AaronMBMorse** [reply]
@janpio removing `shadowDatabaseUrl` fixed it for me. Thanks
